src/main/java/com/faforever/client/avatar/AvatarService.java

```diff
@@ -29,12 +33,26 @@
   private final AssetService assetService;
   private final EventBus eventBus;
   private final AvatarMapper avatarMapper;
+  private final Map<String, AvatarBean> avatarsByPlayerName = new ConcurrentHashMap<>();
 
   @Cacheable(value = AVATARS, sync = true)
   public Image loadAvatar(AvatarBean avatar) {
+    if (avatar == null) {
+      return null;
+    }
     return assetService.loadAndCacheImage(avatar.getUrl(), Path.of("avatars"), null);
   }
 
+  public Optional<Image> getCurrentAvatarByPlayerName(String playerName) {
+    return Optional.ofNullable(loadAvatar(avatarsByPlayerName.get(playerName)));
+  }
+
+  public void addPlayerAvatarToMap(PlayerBean playerBean) {
+    if (playerBean.getAvatar() != null) {
+      avatarsByPlayerName.putIfAbsent(playerBean.getUsername(), playerBean.getAvatar());
+    }
+  }
+
```

Comment:
Right now this completely avoids the current caching system which uses the @ Cacheable annotation. 
As put here this will load the image from the file system everytime the getCurrentAvatarByPlayerName is called. Rather than storing your own cache map of string to AvatarBean you should probably use the playerService to get the playerBean by player name and then get the avatarBean from the player bean and load it using the loadAvatar function.

The other issue with this method is that the avatar is not updated in the map you have if the player changes avatar while online. and it wouldn't even get updated if the player logs out and back in.

As it stands this is just a little redundant information and causes too much coupling.

Reply:
>  you should probably use the playerService to get the playerBean by player name
You forget about performance. What if the `playersByName` map contains 10000, 100,000, or more elements, of which 1000 players have avatars? Calling `map.get()` method for the map will be expensive. We still need a separate map.

Reply:
it doesn't matter how many players are inside of a map. Calling get on the hash map is a O(1) operation. It is the same speed no matter how large the map is.

Reply:
Thanks very much for explaining it.